### PR TITLE
Diffrot for maps - accepts quantity and time (and kwargs!)

### PR DESCRIPTION
**Pull request by dpshelio**
This is just a PR so I pass @ehsteve on the numbers of commits for 0.8

Sure, it also adds some functionality and closes some issues :)

### CODE REVIEW

**Comment by nabobalis** [comment]
Closes #2252

File: sunpy/physics/differential_rotation.py [hunk 1]
```diff
@@ -254,7 +256,15 @@ def diffrot_map(smap, dt, pad=False):
         A map with the result of applying solar differential rotation to the
         input map.
     """
-    new_time = smap.date + timedelta(seconds=dt.to(u.s).value)
+    if time and dt:
```
Comment (wafels) [inline sunpy/physics/differential_rotation.py]: Should this not be more explicit....
if (time is not None) and (dt is not None):

Comment (nabobalis) [inline sunpy/physics/differential_rotation.py]: Thats a good point, we tend to be explicit here with our Nones.

**Comment by nabobalis** [comment]
Merge 3/X

**Pull request merged**